tests/test_java.py

```diff
@@ -156,3 +157,30 @@ def test_java_async_profiler_musl_and_cpu(
         assert_function_in_collapsed(
             "do_syscall_64_[k]", "java", process_collapsed, True
         )  # ensure kernels stacks exist
+
+
+@pytest.mark.parametrize("in_container", [False])
+@pytest.mark.parametrize("check_app_exited", [False])
+def test_hotspot_error_file(application_process, tmp_path, monkeypatch, caplog):
+    start_async_profiler = AsyncProfiledProcess.start_async_profiler
+
+    # Simulate crashing process
+    def sap_and_crash(self, *args, **kwargs):
+        result = start_async_profiler(self, *args, **kwargs)
+        application_process.send_signal(signal.SIGBUS)
+        # Note: unless we wait here the process remains "defunct" and psutil reports it as still running.
```

Comment:
Perhaps we should change the code from `is_running()` to `is_running() and not zombie` like in https://github.com/Granulate/gprofiler/blob/8809b513d08acc50b7b7c023c5af929b87ad78cc/gprofiler/main.py#L353 ? can extract it into some util function.

Reply:
done